mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -317,23 +317,51 @@ class DotConverter final : public OpConversionPattern<migraphx::DotOp> {
 
     // check batch dimension. Tosa matmul only allow a single dimension for it,
     // add reshape ops to flatten and restore the original dimension.
-    ArrayRef<int64_t> orgOutDims = outputTy.getShape();
+    SmallVector<int64_t, 5> orgOutDims(outputTy.getShape());
```

Comment:
Out of curiosity, why cant we get away with a ArrayRef here (as before)?

Reply:
Right.
In the earlier version of this change tried to modify this array. Now it's not the case and I should revert this back, thanks.